Restrict WebView navigation to themoviedb.org

MostafaMohamed2002 opened the pull request:
Updated the WebView component to block navigation to external domains, allowing only internal navigation within www.themoviedb.org. Displays a toast message when navigation is blocked.<br>

https://github.com/user-attachments/assets/c01ff116-6267-4242-acbb-014621524318

MostafaMohamed2002 committed:
Restrict WebView navigation to themoviedb.org

Updated the WebView component to block navigation to external domains, allowing only internal navigation within www.themoviedb.org. Displays a toast message when navigation is blocked.

MennaSrour commented on design_system/src/main/java/com/cairosquad/design_system/basic_component/WebView.kt:
```diff
@@ -36,11 +38,21 @@ fun WebView(
+                    override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
+                        val uri = Uri.parse(url)
+                        return if (uri.host == "www.themoviedb.org") {
+                            false // allow internal navigation
```
you can remove the comment if it isn't import

MennaSrour reviewed (approved):
good work

MostafaMohamed2002 committed:
Restrict WebView navigation to allowed TMDB paths

Updated the WebView component to block navigation to URLs outside specific allowed paths and authentication routes on www.themoviedb.org. Displays a toast message when navigation is blocked for improved user feedback.

MostafaMohamed2002 committed:
Removed '/login' and '/authenticate' from the list of allowed TMDB paths
